metamist/parser/generic_metadata_parser.py

```diff
@@ -257,18 +257,33 @@ def get_read_length(self, row: SingleRow) -> int:
         value = int(value) if value else None
         return value
 
-    def get_assay_id(self, row: GroupedRow) -> Optional[dict[str, str]]:
+    def get_assay_id(self, row: GroupedRow) -> dict[str, str] | None:
         """Get external assay ID from row. Needs to be implemented per parser.
         NOTE: To be re-thought after assay group changes are applied"""
         return None
 
-    def get_participant_id(self, row: SingleRow) -> Optional[str]:
+    def get_assay_meta(self, row: GroupedRow) -> dict[str, Any] | None:
```

Comment:
I think this typing is wrong, you're typing it as a GroupedRow (list of dicts), but you're treating it as a SingleRow (and it's called as a Single Row). IMO, it should be the GroupedRow, or instead of this, use the `self.collapse_arbitrary_meta` function instead.